### PR TITLE
[MRG] fix py3k errors + reorg tests + some more

### PR DESCRIPTION
**Pull request by agramfort**

### CODE REVIEW

File: autoreject/ransac.py [hunk 1]
```diff
@@ -102,12 +95,13 @@ def __init__(self, n_resample=50, min_channels=0.25, min_corr=0.75,
         self.min_corr = min_corr
         self.unbroken_time = unbroken_time
         self.n_jobs = n_jobs
+        self.random_state = random_state
         self.verbose = verbose
 
     def _get_random_subsets(self, info):
         """ Get random channels"""
         # have to set the seed here
-        np.random.seed(435656)
```
Comment (agramfort) [inline autoreject/ransac.py]: @jasmainak global seeding is dangerous

File: autoreject/ransac.py [hunk 2]
```diff
@@ -116,7 +110,7 @@ def _get_random_subsets(self, info):
         # get picks for resamples
         picks = []
         for idx in range(self.n_resample):
-            pick = _randsample(range(n_channels), n_samples)
+            pick = rng.permutation(n_channels)[:n_samples].copy()
```
Comment (agramfort) [inline autoreject/ransac.py]: @jasmainak I find it much simpler ok?

File: autoreject/ransac.py (hunk 1 above)
Comment (jasmainak) [inline autoreject/ransac.py]: yeah actually I had literally translated the Matlab code ...

File: autoreject/utils.py [hunk 3]
```diff
@@ -121,8 +125,8 @@ def fetch_file(url, file_name, resume=True, timeout=10.):
         The URL open timeout.
     """
     from mne.utils import _fetch_file
-    return _fetch_file(url=url, file_name=file_name, print_destination=True,
-                       resume=resume, hash_=None, timeout=timeout)
```
Comment (agramfort) [inline autoreject/utils.py]: nothing is returned

File: examples/plot_ransac.py [hunk 4]
```diff
@@ -65,10 +65,10 @@
 
 ###############################################################################
 # We import ``Ransac`` and run the familiar ``fit_transform`` method.
-from autoreject import Ransac
-from autoreject.utils import interpolate_bads
+from autoreject import Ransac  # noqa
+from autoreject.utils import interpolate_bads  # noqa
 
-ransac = Ransac(verbose='tqdm')
+ransac = Ransac(verbose='progressbar', n_jobs=1)
```
Comment (agramfort) [inline examples/plot_ransac.py]: I don't have tqdm. Don't make it a hard dependency.

Comment (jasmainak) [inline examples/plot_ransac.py]: It's not a hard dependency. It was just in the example to illustrate ... but okay for changing it to `'progressbar'`

File: autoreject/ransac.py [hunk 5]
```diff
@@ -13,21 +13,12 @@
...
 from .autoreject import _check_data
 
 
-def _randsample(X, num):
-    """Generate random subsamples (as done in PREP code)"""
-    Y = []
-    for k in range(num):
-        pick = int(round(1 + (len(X) - 1) * np.random.random())) - 1
-        Y.append(X[pick])
-        del X[pick]
```
Comment (agramfort) [inline autoreject/ransac.py]: @jasmainak this was failing on py3k as range is a generator.

Comment (jasmainak) [inline autoreject/ransac.py]: ah okay got it, thanks!

File: examples/plot_ransac.py (hunk 4 above)
Comment (agramfort) [inline examples/plot_ransac.py]: then introduce an "auto" mode that defaults to tqdm if available

**Comment by agramfort** [comment]
I am done here. @jasmainak travis is finally happy

just maybe check the pb of the random_state in Ransac

File: examples/plot_visualize_bad_epochs.py [hunk 6]
```diff
@@ -29,7 +26,7 @@
 subject = "sub%03d" % subject_id
 
 print("processing subject: %s" % subject)
-base_path = os.path.join(os.path.dirname(autoreject.__file__), '..', 'examples')
+base_path = os.path.join(os.path.dirname(__file__))
```
Comment (jasmainak) [inline examples/plot_visualize_bad_epochs.py]: @agramfort you broke this. I can't build the doc for this file now ...

**Comment by jasmainak** [comment]
Otherwise LGTM. random_state seems to be fine. I get the same output on every run

File: examples/plot_visualize_bad_epochs.py (hunk 6 above)
Comment (agramfort) [inline examples/plot_visualize_bad_epochs.py]: reverted

**Comment by agramfort** [comment]
@jasmainak we should be good here

**Comment by jasmainak** [comment]
@agramfort I was just building the docs. Looks good. Thanks for the fixes!

File: examples/plot_visualize_bad_epochs.py [hunk 7]
```diff
@@ -83,36 +79,34 @@
...
-from autoreject import LocalAutoRejectCV, compute_thresholds
-from functools import partial
+from autoreject import LocalAutoRejectCV, compute_thresholds  # noqa
+from functools import partial  # noqa
 
 this_epoch = epochs['famous']
 thresh_func = partial(compute_thresholds, random_state=42)
 
-ar = LocalAutoRejectCV(thresh_func=thresh_func, verbose='tqdm')
+ar = LocalAutoRejectCV(thresh_func=thresh_func, verbose='progressbar')
```
Comment (jasmainak) [inline examples/plot_visualize_bad_epochs.py]: arff ... now I remember why I had tqdm here. There is something buggy about progressbar which is why `circle` fails on `master`. No output for more than 10 minutes ...

And circle isn't showing up on PRs any more for some reason.

**Pull request merged**